policy/modules/system/systemd.te

```diff
@@ -1132,3 +1136,42 @@ files_var_filetrans(systemd_update_done_t, systemd_update_run_t, file, ".updated
 seutil_read_file_contexts(systemd_update_done_t)
 
 systemd_log_parse_environment(systemd_update_done_t)
+
+#########################################
+#
+# User session (systemd --user) local policy
+#
+
+allow systemd_user_session_type self:capability { dac_read_search sys_resource };
+dontaudit systemd_user_session_type self:capability dac_override;
+allow systemd_user_session_type self:process setfscreate;
+allow systemd_user_session_type self:udp_socket create_socket_perms;
+allow systemd_user_session_type self:unix_stream_socket create_stream_socket_perms;
+allow systemd_user_session_type self:netlink_kobject_uevent_socket { bind create getattr setopt };
+
+allow systemd_user_session_type systemd_user_runtime_t:dir manage_dir_perms;
+allow systemd_user_session_type systemd_user_runtime_t:sock_file { create write };
```

Comment:
Not sure if you want this added complexity but systemd maintains two sockets, for two different purposes.. "private" and "notify", With a name-based type transttion you would be able to diferentiate between the two.

Reply:
I am aware of them. I'm open to having them separated in a future change.

Reply:
I take that back. If you elaborate on this point, I could add it in this PR.

Reply:
This might explain a bit about the notify socket:

https://www.freedesktop.org/software/systemd/man/systemd-notify.html

So just because a service unit may have Type=notify or just because systemd-notify is used to notify systemd of status changes, that does not mean that these processes also have to connecto to systemd with the private socket

Reply:
Are you suggesting three types, one for each socket plus the general `systemd_user_runtime_t`, or would it be sufficient to do a separate type just for notify?

Reply:
I just have a private type "systemd_user_runtime_private_t" for the "private" socket. The notify socket is just generic systemd_user_runtime_t just as the directory. That would be enough to differentiate between the notify socket and the private socket.

Reply:
I would have thought the notify socket would be the one with the different type, since that would theoretically be the one other domains interact with.

Reply:
```
$ ls -Z /run/user/1000/systemd/*
       wheel.id:wheel.role:systemd.user.tmpfs.user_tmpfs_file:s0 /run/user/1000/systemd/notify
     wheel.id:wheel.role:systemd.user.private.user_tmpfs_file:s0 /run/user/1000/systemd/private

/run/user/1000/systemd/transient:
wheel.id:wheel.role:systemd.user.unit_runtime.user_tmpfs_file:s0 dbus-:1.2-ca.desrt.dconf@0.service
wheel.id:wheel.role:systemd.user.unit_runtime.user_tmpfs_file:s0 dbus-:1.8-org.a11y.atspi.Registry@0.service

```
So something like this:

1. /run/user/UID/systemd:  systemd_user_runtime_t
2. /run/user/UID/systemd/notify: systemd_user_runtime_t
3. /run/user/UID/systemd/private: systemd_user_runtime_private_t
4. /run/user/UID/systemd/transient(/.*)? systemd_user_runtime_unit_t

Reply:
or simplified:

    /run/user/UID/systemd(/.*)?: systemd_user_runtime_t
    /run/user/UID/systemd/private: systemd_user_runtime_private_t
    /run/user/UID/systemd/transient(/.*)? systemd_user_runtime_unit_t

Reply:
> I would have thought the notify socket would be the one with the different type, since that would theoretically be the one other domains interact with.

I suppose. I dont think it matter much. Just as long as you can differentiate between private and notify.

Reply:
But yes I you like this better:

```
/run/user/UID/systemd(/.*)? systemd_user_runtime_t
/run/user/UID/systemd/notify systemd_user_runtime_notify_t
/run/user/UID/systemd/transient(/.*)? systemd_user_runtime_unit_t
```